arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -377,8 +329,33 @@ int arm_core_mpu_get_max_domain_partition_regions(void)
 		_get_region_index_by_type(THREAD_DOMAIN_PARTITION_REGION);
 }
 
+/**
+ * This internal function checks if the region is user accessible or not
+ */
+static inline int _is_user_accessible_region(u32_t r_index, int write)
+{
+	u32_t r_ap;
+
+	ARM_MPU_DEV->rnr = r_index;
+	r_ap = ARM_MPU_DEV->rasr & ACCESS_PERMS_MASK;
+
+	/* always return true if this is the thread stack region */
+	if (_get_region_index_by_type(THREAD_STACK_REGION) == r_index) {
```

Comment:
Why this condition is needed?  The `r_ap == P_RW_U_RW` check will fail there?

Reply:
This might be a throwback to when the stack region index was the same for the privileged and unprivileged stack.  The permissions on those are different and I believe this check would fail because it was stricter for the privileged stack.  With the changes done to split out the privileged stack, we don't have to worry about the locality of the stack as it's in a completely different memory region and any user mode access to that memory would fail.

Is this function ever called when the thread is running privileged mode?